src/Umbraco.Core/UdiEntityType.cs

```diff
@@ -13,81 +13,93 @@ public static partial class Constants
         /// but entity types are strings and so can be extended beyond what is defined here.</remarks>
         public static class UdiEntityType
         {
-            [UdiType(UdiType.Unknown)]
+            // note: const fields in this class MUST be consistent with what GetTypes returns
+            //  this is validated by UdiTests.ValidateUdiEntityType
+
+            internal static Dictionary<string, UdiType> GetTypes()
+            {
+                return new Dictionary<string,UdiType>
```

Comment:
@zpqrtbnk shouldn't this be stored in a field so we aren't re-allocating a new Dictionary object every time GetTypes() is called? And this should be ReadOnlyDictionary since it shouldn't be mutable.

Reply:
it's internal and meant to be used by `Udi` static initialized only, we probably don't want to keep the object around in a field. and then, does not really matter if it's readonly?

Reply:
ah ok, i was just looking at this code - not how it's used - based on that this is why I've left these comments

Reply:
maybe some code comments would help in that case